osu.Game/Overlays/SkinEditor/SkinBlueprint.cs

```diff
@@ -100,82 +115,90 @@ protected override void OnDeselected()
-
-        public override Quad SelectionQuad => drawable.ScreenSpaceDrawQuad;
     }
 
     internal partial class AnchorOriginVisualiser : CompositeDrawable
     {
         private readonly Drawable drawable;
 
-        private readonly Box originBox;
+        private Drawable originBox = null!;
 
-        private readonly Box anchorBox;
-        private readonly Box anchorLine;
+        private Drawable anchorBox = null!;
+        private Drawable anchorLine = null!;
 
         public AnchorOriginVisualiser(Drawable drawable)
         {
             this.drawable = drawable;
+        }
 
-            InternalChildren = new Drawable[]
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            Color4 anchorColour = colours.Red1;
+            Color4 originColour = colours.Red3;
+
+            InternalChildren = new[]
             {
-                anchorLine = new Box
+                anchorLine = new Circle
                 {
-                    Height = 2,
+                    Height = 3f,
                     Origin = Anchor.CentreLeft,
-                    Colour = Color4.Yellow,
-                    EdgeSmoothness = Vector2.One
+                    Colour = ColourInfo.GradientHorizontal(originColour.Opacity(0.5f), originColour),
                 },
-                originBox = new Box
+                originBox = new Circle
                 {
-                    Colour = Color4.Red,
+                    Colour = originColour,
                     Origin = Anchor.Centre,
-                    Size = new Vector2(5),
+                    Size = new Vector2(7),
                 },
-                anchorBox = new Box
+                anchorBox = new Circle
                 {
-                    Colour = Color4.Red,
+                    Colour = anchorColour,
                     Origin = Anchor.Centre,
-                    Size = new Vector2(5),
+                    Size = new Vector2(10),
                 },
             };
         }
 
+        private Vector2? anchorPosition;
+
         protected override void Update()
         {
             base.Update();
 
             if (drawable.Parent == null)
                 return;
 
+            var newAnchor = drawable.Parent.ToSpaceOfOtherDrawable(drawable.AnchorPosition, this);
+
+            anchorPosition ??= newAnchor;
+
+            anchorPosition =
+                new Vector2(
+                    (float)Interpolation.DampContinuously(anchorPosition.Value.X, newAnchor.X, 25, Clock.ElapsedFrameTime),
+                    (float)Interpolation.DampContinuously(anchorPosition.Value.Y, newAnchor.Y, 25, Clock.ElapsedFrameTime)
+                );
+
             originBox.Position = drawable.ToSpaceOfOtherDrawable(drawable.OriginPosition, this);
-            anchorBox.Position = drawable.Parent.ToSpaceOfOtherDrawable(drawable.AnchorPosition, this);
+            anchorBox.Position = anchorPosition.Value;
```

Comment:
Seems weird that the anchor position tweens, but the origin position does not:

https://user-images.githubusercontent.com/20418176/219792561-012c375a-34d3-4033-86f8-13d659d5407a.mp4

I'm putting aside the fact that the origin position circle is mostly obscured by the selection box (which isn't exactly new, so I'm willing to let it slide in this pull).

Reply:
I wanted to leave it attached to the thing the user is dragging, which was the main case I was focused on.

I agree it would be nice to have it tweet when changing the setting from the context menu while not dragging, but not sure how simple that will be to make happen..

Reply:
Right, I see what you're driving at. I thought this was going to be a bit simpler than it is. That said, it can be done if the damping is applied in drawable space rather than screen space. Something like so:

```diff
diff --git a/osu.Game/Overlays/SkinEditor/SkinBlueprint.cs b/osu.Game/Overlays/SkinEditor/SkinBlueprint.cs
index 2cb14814d1..f5814b4c01 100644
--- a/osu.Game/Overlays/SkinEditor/SkinBlueprint.cs
+++ b/osu.Game/Overlays/SkinEditor/SkinBlueprint.cs
@@ -177,6 +177,7 @@ private void load(OsuColour colours)
         }
 
         private Vector2? anchorPosition;
+        private Vector2? originPositionInDrawableSpace;
 
         protected override void Update()
         {
@@ -186,18 +187,13 @@ protected override void Update()
                 return;
 
             var newAnchor = drawable.Parent.ToSpaceOfOtherDrawable(drawable.AnchorPosition, this);
-
-            anchorPosition ??= newAnchor;
-
-            anchorPosition =
-                new Vector2(
-                    (float)Interpolation.DampContinuously(anchorPosition.Value.X, newAnchor.X, 25, Clock.ElapsedFrameTime),
-                    (float)Interpolation.DampContinuously(anchorPosition.Value.Y, newAnchor.Y, 25, Clock.ElapsedFrameTime)
-                );
-
-            originBox.Position = drawable.ToSpaceOfOtherDrawable(drawable.OriginPosition, this);
+            anchorPosition = tweenPosition(anchorPosition ?? newAnchor, newAnchor);
             anchorBox.Position = anchorPosition.Value;
 
+            // for the origin, tween in the drawable's local space to avoid unwanted tweening when the drawable is being dragged.
+            originPositionInDrawableSpace = originPositionInDrawableSpace != null ? tweenPosition(originPositionInDrawableSpace.Value, drawable.OriginPosition) : drawable.OriginPosition;
+            originBox.Position = drawable.ToSpaceOfOtherDrawable(originPositionInDrawableSpace.Value, this);
+
             var point1 = ToLocalSpace(anchorBox.ScreenSpaceDrawQuad.Centre);
             var point2 = ToLocalSpace(originBox.ScreenSpaceDrawQuad.Centre);
 
@@ -205,5 +201,11 @@ protected override void Update()
             anchorLine.Width = (point2 - point1).Length;
             anchorLine.Rotation = MathHelper.RadiansToDegrees(MathF.Atan2(point2.Y - point1.Y, point2.X - point1.X));
         }
+
+        private Vector2 tweenPosition(Vector2 oldPosition, Vector2 newPosition)
+            => new Vector2(
+                (float)Interpolation.DampContinuously(oldPosition.X, newPosition.X, 25, Clock.ElapsedFrameTime),
+                (float)Interpolation.DampContinuously(oldPosition.Y, newPosition.Y, 25, Clock.ElapsedFrameTime)
+            );
     }
 }
```

https://user-images.githubusercontent.com/20418176/219868473-6acadd33-1e41-4515-a1c5-c8bb4b052ced.mp4

Reply:
Looking at your video only, I think your change may have broken the part where the component's location doesn't change when altering origin:

https://user-images.githubusercontent.com/191335/219954374-e8038ce2-806c-4fb1-9dcd-564e11d8dff6.mp4

Reply:
I don't... think so? I would be pretty flabbergasted if a change local to `AnchorOriginVisualiser` would break this.

As far as I can tell the issue is actually with the legacy circular progress component. I'm seeing the same behaviour of it shifting around with origin on `master`.

Reply:
Ah, weird stuff. That's what I get for not reading the code 😄

Reply:
Seems to work well, I've applied the change.